modules/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageRetryingInputStream.java

```diff
@@ -84,6 +86,30 @@ class GoogleCloudStorageRetryingInputStream extends InputStream {
         currentStream = openStream();
     }
 
+    // Used for testing only
+    GoogleCloudStorageRetryingInputStream(
```

Comment:
nit: I think that the previous constructor could use this one using `this(client, () -> getStorage(client), ....)` if we were passing a Function<Storage, Storage> function?

Reply:
Not a nit. It makes code much cleaner! Thank you